core/app/services/uffizzi_core/compose_file/container_service.rb

```diff
@@ -27,6 +27,12 @@ def docker_hub?(container)
       registry_url.nil? && repository_url.nil?
     end
 
+    def docker_registry?(container)
+      registry_url = container.dig(:image, :registry_url)
+      host_regexp = /(\w+\.\w{2,})(?::\d+)?\z/
+      registry_url.present? && ['amazonaws.com', 'azurecr.io', 'gcr.io', 'ghcr.io'].exclude?(registry_url.match(host_regexp)&.to_a&.last)
```

Comment:
Can we return false if it's not present and does not match the regex first? It would be cleaner.